dealership/contracts.py

```diff
@@ -1,12 +1,54 @@
+from .vehicles import Car, Truck, Motorcycle
+
+
 class Contract(object):
-    pass
+    def __init__(self, vehicle, customer):
+        self.vehicle = vehicle
+        self.customer = customer
+
+    def total_value(self):
+        pass
+
+    def monthly_value(self):
+        pass
 
 
 class BuyContract(Contract):
+    INTEREST_RATES = {
+        Car: 1.07,
+        Motorcycle: 1.03,
+        Truck: 1.11,
+    }
+
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
+        super(BuyContract, self).__init__(vehicle, customer)
+        self.monthly_payments = monthly_payments
+
+    def total_value(self):
+        return (self.vehicle.sale_price() + BuyContract.INTEREST_RATES[type(self.vehicle)]
+                * self.monthly_payments * self.vehicle.sale_price() / 100) * (0.9 if self.customer.is_employee() else 1)
```

Comment:
Small suggestion: Instead of hard-coding the discount rate (`0.9`) when calculating `total_value()` (in `BuyContract` and `LeaseContract`, you could set it as an instance variable in `Vehicle`. Therefore, you will only have to change the rate in one spot if you decide alter the rate. I know changing a hard-coded `0.9` is a very simple fix in the project, but in a setting with a larger number of sub-classes, it may be better practice to find ways where the code only needs to be changed in one spot.